server/container/guice/guice-common/src/main/java/org/apache/james/GuiceJamesServerImpl.java

```diff
@@ -65,11 +73,21 @@ public GuiceJamesServerImpl overrideWith(Module... overrides) {
     @Override
     public void start() throws Exception {
         Injector injector = Guice.createInjector(module);
+        logInjectedElementsWithNoScope(injector);
+
         preDestroy = injector.getInstance(Key.get(new TypeLiteral<Stager<PreDestroy>>() {}));
         injector.getInstance(ConfigurationsPerformer.class).initModules();
         guiceProbeProvider = injector.getInstance(GuiceProbeProvider.class);
     }
 
+    private void logInjectedElementsWithNoScope(Injector injector) {
+        injector.getAllBindings().entrySet().stream()
```

Comment:
Maybe you can reduce the number of logs, but still keep this (soooo) usefull info by only logging when dependancy is injected twice with no scope ?

(IE : when it is truely dangerous ?)